src/execution_tree/compile.cpp

```diff
@@ -134,12 +134,12 @@ namespace phylanx { namespace execution_tree
     ///////////////////////////////////////////////////////////////////////////
     /// Add the given variable to the compilation environment
     compiler::function define_variable(std::string const& codename,
-        std::string name, compiler::function_list& snippets,
-        compiler::environment& env, primitive_argument_type body,
-        hpx::id_type const& default_locality)
+        compiler::primitive_name_parts const& name_parts,
+        compiler::function_list& snippets, compiler::environment& env,
+        primitive_argument_type body, hpx::id_type const& default_locality)
     {
         return compiler::define_variable(
-            codename, std::move(name), snippets, env, body, default_locality)();
+            codename, std::move(name_parts), snippets, env, body, default_locality)();
```

Comment:
The `std::move` is possibly moving from a const object - not good.

Reply:
Fixed.